pkg/server/plugin/upstreamauthority/spire/spire_test.go

```diff
@@ -215,8 +216,18 @@ func (h *handler) Attest(stream node_pb.Node_AttestServer) (err error) {
 	return errors.New("NOT IMPLEMENTED")
 }
 
+// PushJWTKeyUpstream fakes the real implementation (node endpoint) for testing purposes
 func (h *handler) PushJWTKeyUpstream(ctx context.Context, req *node_pb.PushJWTKeyUpstreamRequest) (*node_pb.PushJWTKeyUpstreamResponse, error) {
-	return nil, errors.New("NOT IMPLEMENTED")
+	keys := []*common.PublicKey{
```

Comment:
maybe you can add an erorr to handler and veritfy if it is nil or return it, so we can test scenaries where PushJWTKeyUpstream back with errors

Reply:
I was not able to find an elegant way to force an error here. But I believe a non-elegant test is better than no test. Will fix

Reply:
I'd be ok with you punting on this. #1452 is going to complicate this plugin and probably require some non-trivial changes to the tests. It can be tackled then....